flamedisx/block_source.py

```diff
@@ -459,6 +587,50 @@ def _annotate(self, _skip_bounds_computation=False):
         for b in self.model_blocks[::-1]:
             b.annotate(d)
 
+    def scale_observables(self, df, observables):
```

Comment:
The `scale_observables` and `energy_bounds` methods are currently unused (if I understood correctly) and undocumented. I understand you'll want to propose them in the next PR, happy to take a look at them then -- in the meantime, could you move them somewhere else or remove them?

For the KDE, any reason not to use scipy's KDE? I think it is [faster](https://kdepy.readthedocs.io/en/latest/comparison.html) and would avoid adding scikit-learn as a dependency to flamedisx.

Reply:
Done; it was a k-d tree, not KDE, but yeah, I'm using the SciPy version now. I moved the `energy_bounds` stuff to bounds.py also; hopefully this is fine